core/src/main/scala/package.scala

```diff
@@ -57,6 +55,12 @@ package object nelson {
   type DNSName = String
   type DeploymentStatusString = String
 
+/** Copied and adapted from Scalaz's Tag implementation.
+  * https://github.com/scalaz/scalaz/blob/v7.1.17/core/src/main/scala/scalaz/package.scala
+  */
+  private[this] type Tagged[A, T] = { type Tag = T; type Self = A; }
```

Comment:
I'm curious why this doesn't exist in `cats`. Is it something that should be added?

Reply:
There's been a couple competing implementations and nobody has converged on something yet, I think at this point a lot of folks are waiting for [SIP-35 opaque types](https://docs.scala-lang.org/sips/opaque-types.html) to land.